### PR TITLE
chore!: txsim to use signer

### PR DESCRIPTION
**Pull request by cmwaters**
## Overview

txsim now uses the `Signer` class. This means txsim no longer requires RPC endpoints, relying solely on gRPC

## Checklist

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### CODE REVIEW

**Comment by rootulp** [comment]
It looks like the test workflow fails b/c a test is attempting to use the `--rpc-endpoints` flag which is removed in this PR. See [here](https://github.com/celestiaorg/celestia-app/actions/runs/5952648552/job/16145071957?pr=2326#step:4:35).

**Review by rootulp — COMMENTED** [review]
LGTM after test passes

**Review by evan-forbes — APPROVED** [review]
LGTM :+1: :+1:

**Review by rootulp — APPROVED** [review]

**Review by evan-forbes — APPROVED** [review]

File: test/cmd/txsim/cli.go [hunk 1]
```diff
@@ -33,13 +34,13 @@ const (
...
 // Values for all flags
 var (
-	keyPath, masterAccName, keyMnemonic, rpcEndpoints, grpcEndpoints string
-	blobSizes, blobAmounts                                           string
-	seed                                                             int64
-	pollTime                                                         time.Duration
-	send, sendIterations, sendAmount                                 int
-	stake, stakeValue, blob                                          int
-	useFeegrant                                                      bool
+	keyPath, masterAccName, keyMnemonic, grpcEndpoint string
```
[the remark this answers is not in the document]
Comment (cmwaters) [reply test/cmd/txsim/cli.go:37]: Good catch. Will do it in a follow up PR

**Pull request merged**